src/core/lv_refr.c

```diff
@@ -1380,6 +1381,11 @@ static void draw_buf_flush(lv_display_t * disp)
     if(disp->flush_cb) {
         call_flush_cb(disp, &disp->refreshed_area, layer->draw_buf->data);
     }
+
```

Comment:
@kisvegabor what if the wait for swapping is moved under the swap condition?

```suggestion
    /*If there are 2 buffers swap them. With direct mode swap only on the last area*/
    if(lv_display_is_double_buffered(disp) && (disp->render_mode != LV_DISPLAY_RENDER_MODE_DIRECT || flushing_last)) {
        wait_for_swapping(disp);
        if(disp->buf_act == disp->buf_1) {
            disp->buf_act = disp->buf_2;
        }
```

I believe that the flush time will not be reported correctly if `disp_event_cb` is used for this.

Reply:
My concern is that I want to introduce a new buffer manager callback only if the current event and call back system is not suitable. 

Is there any way to use the currently available tools?